docker/prognostic_run/Dockerfile

```diff
@@ -4,44 +4,29 @@ FROM ${BASE_IMAGE} as bld
 
 # build/install the fortran model
 COPY external/fv3gfs-fortran/ /tmp/fortran-build
-RUN cd /tmp/fortran-build/FV3 && \
-    ./configure gnu_docker && \
-    sed -i "33i CALLPYFORT=Y" conf/configure.fv3 && \
-    make -j 8 && \
-    PREFIX=/usr/local make install
-
-# Python Stuff Here
-#
-# Equivalent to passing --no-cache-dir to every pip install
-ENV PIP_NO_CACHE_DIR=off
-
-# Install dependencies
-COPY docker/prognostic_run/requirements.txt /tmp/requirements.txt
-RUN pip install -r /tmp/requirements.txt
-
-# compile and install the wrapper. Then import fv3gfs.wrapper as a minimal test
-# that it works.
-RUN make -C /tmp/fortran-build/FV3 wrapper_build &&\
-    pip3 install --no-dependencies /tmp/fortran-build/FV3/wrapper/dist/*.whl && \
-    python3 -c 'import fv3gfs.wrapper'
+COPY .environment-scripts/install_fv3gfs_fortran.sh .
+RUN CALLPYFORT=Y bash install_fv3gfs_fortran.sh /tmp/fortran-build/FV3 gnu_docker /usr/local
 
+# Install the Python dependencies
+COPY docker/prognostic_run/requirements.txt /fv3net/docker/prognostic_run/requirements.txt
 COPY external/vcm /fv3net/external/vcm
-RUN pip3 install --no-dependencies -e /fv3net/external/vcm
-
 COPY external/artifacts /fv3net/external/artifacts
-RUN pip3 install --no-dependencies -e /fv3net/external/artifacts
-
 COPY external/loaders /fv3net/external/loaders
-RUN pip3 install --no-dependencies -e /fv3net/external/loaders
-
 COPY external/fv3fit /fv3net/external/fv3fit
-RUN pip3 install --no-dependencies -e /fv3net/external/fv3fit
-
 COPY external/fv3kube /fv3net/external/fv3kube
-RUN pip3 install --no-dependencies -e /fv3net/external/fv3kube
-
 COPY workflows/post_process_run /fv3net/workflows/post_process_run
-RUN pip3 install --no-dependencies -e /fv3net/workflows/post_process_run
+COPY workflows/prognostic_c48_run/ /fv3net/workflows/prognostic_c48_run
+COPY external/emulation /fv3net/external/emulation
+COPY projects/microphysics/scripts /fv3net/projects/microphysics/scripts
+COPY external/radiation /fv3net/external/radiation
+COPY .environment-scripts/install_fv3net_python_dependencies.sh .
+RUN bash install_fv3net_python_dependencies.sh /fv3net
```

Comment:
It's now not obvious that two things need to be modified for adding 'local' python dependencies (i.e. add COPY to Dockerfile and pip install to shell script).

Could we have an variable that is the list of fv3net packages to be pip installed which gets passed to the script?

Reply:
Implemented in [da8fb0b](https://github.com/ai2cm/fv3net/pull/2035/commits/da8fb0bf2ba5e7a99ac10b0f5002cdab1b9c218c).